class-gf-razorpay.php

```diff
@@ -0,0 +1,285 @@
+            ),
+            'order_id'    => $entry['razorpay_order_id'],
+        );
+
+        $json = $razorpay_args;
+
+        $redirect_url = '?page=gf_razorpay_callback';
+
+
+        wp_enqueue_script('razorpay-script', plugin_dir_url( __FILE__ ) . 'script.js',
+            array('checkout'));
+
+        wp_localize_script('razorpay-script', 'razorpay_script_vars', array(
+            'data' => $json
+        ));
+
+        wp_register_script( 'checkout', 'https://checkout.razorpay.com/v1/checkout.js', null, null);
+
+        wp_enqueue_script('checkout');
+
+        return $this->generate_order_form($redirect_url);
+    }
+
+    function generate_order_form($redirect_url)
+    {
+        $html = <<<EOT
+<form id ='razorpayform' name='razorpayform' action="$redirect_url" method='POST'>
+    <input type='hidden' name='razorpay_payment_id' id='razorpay_payment_id'>
+    <input type='hidden' name='razorpay_signature'  id='razorpay_signature' >
+</form>
+<p id='msg-razorpay-success'  style='display:none'>
+    Please wait while we are processing your payment.
+</p>
+<p>
+    <button id='btn-razorpay'>Pay Now</button>
+    <button id='btn-razorpay-cancel' onclick='document.razorpayform.submit()'>Cancel</button>
+</p>
+EOT;
+        return $html;
+    }
+
+    public function is_callback_valid()
+    {
+        if (rgget( 'page' ) !== 'gf_razorpay_callback')
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    public function pay_using_razorpay($entry, $form)
+    {
+        $feed = $this->get_payment_feed($entry, $form);
+
+        $customer_fields = $this->get_customer_fields_array($feed, $entry);
+
+        $payment_amount = rgar( $entry, 'payment_amount' );
+
+        //It will be null first time in the entry
+        if (empty($payment_amount) === true)
+        {
+            $payment_amount = GFCommon::get_order_total($form, $entry);
+            gform_update_meta( $entry['id'], 'payment_amount', $payment_amount );
+            $entry['payment_amount'] = $payment_amount;
+        }
+
+        $api = new Api($this->get_plugin_setting('gf_razorpay_key'),
+            $this->get_plugin_setting('gf_razorpay_secret'));
+
+        $data = array(
+            'receipt'         => $entry['id'],
+            'amount'          => (int) round($payment_amount * 100),
+            'currency'        => $entry['currency'],
+            'payment_capture' => 1
+        );
+
+        $razorpay_order = $api->order->create($data);
+
+        gform_update_meta($entry['id'], 'razorpay_order_id', $razorpay_order['id']);
+
+        $entry['razorpay_order_id'] = $razorpay_order['id'];
+
+        GFAPI::update_entry($entry);
+
+        //cookie set for one day
+        setcookie('entry_id', $entry['id'],
+            time() + 86400, COOKIEPATH, COOKIE_DOMAIN, false, true);
+        setcookie('razorpay_order_id', $entry['razorpay_order_id'],
+            time() + 86400, COOKIEPATH, COOKIE_DOMAIN, false, true);
+
+
```

Comment:
1. Where are `COOKIEPATH` and `COOKIE_DOMAIN` defined?
2. Set the secure flag to true if the website is served over HTTPS.